base/bucket_n1ql_test.go

```diff
@@ -0,0 +1,247 @@
+package base
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/couchbase/gocb"
+	"github.com/couchbaselabs/go.assert"
+)
+
+func TestN1qlQuery(t *testing.T) {
+
+	if UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server")
+	}
+
+	testBucket := GetTestBucketOrPanic()
+	defer testBucket.Close()
+	bucket, ok := testBucket.Bucket.(*CouchbaseBucketGoCB)
+	if !ok {
+		t.Errorf("Requires gocb bucket")
```

Comment:
We probably want to use `t.Fatalf` here to stop the rest of the test executing.

Reply:
Strictly speaking this is already covered by the walrus skip, but good call to add the additional check.